components/tidb_query_aggr/src/util.rs

```diff
@@ -39,10 +39,20 @@ pub fn rewrite_exp_for_sum_avg(schema: &[FieldType], exp: &mut RpnExpression) ->
             // No need to cast. Return directly without changing anything.
             return Ok(());
         }
-        EvalType::Int => FieldTypeBuilder::new()
-            .tp(FieldTypeTp::NewDecimal)
-            .flen(tidb_query_datatype::MAX_DECIMAL_WIDTH)
-            .build(),
+        EvalType::Int => {
+            if ret_field_type.tp() == FieldTypeTp::Bit {
+                FieldTypeBuilder::new()
+                    .tp(FieldTypeTp::Double)
```

Comment:
Why is `Bit`'s field type `Double` in this part? Is this specified in TiDB?

Reply:
Codes tidb\expression\aggregation\base_func.go  define typeInfer4Sum()、typeInfer4Avg(), For Bit RetType is Double.
TiDB's corresponding PR:  https://github.com/pingcap/tidb/pull/31884